src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,511 @@
+
+    if (transportConfiguration == nullptr)
+    {
+        ChipLogError(Zcl, "HandleManuallyTriggerTransport: ConnectionID Not Found.");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    if (transportConfiguration->transportStatus == TransportStatusEnum::kInactive)
+    {
+        auto clusterStatus = static_cast<uint8_t>(StatusCodeEnum::kInvalidTransportStatus);
+        ChipLogError(Zcl, "HandleManuallyTriggerTransport: Invalid Transport status");
+        ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, clusterStatus);
+        return;
+    }
+    if (transportConfiguration->transportOptions.HasValue())
+    {
+        if (transportConfiguration->transportOptions.Value().triggerOptions.triggerType == TransportTriggerTypeEnum::kContinuous)
+        {
+            {
+                auto clusterStatus = static_cast<uint8_t>(StatusCodeEnum::kInvalidTriggerType);
+                ChipLogError(Zcl, "HandleManuallyTriggerTransport: Invalid Trigger type");
+                ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, clusterStatus);
+                return;
+            }
+        }
+        else if (transportConfiguration->transportOptions.Value().triggerOptions.triggerType ==
+                     TransportTriggerTypeEnum::kCommand &&
+                 !timeControl.HasValue())
+        {
+            {
+                ChipLogError(Zcl, "HandleManuallyTriggerTransport: Time control field not present");
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+                return;
+            }
+        }
+    }
+
+    // Call the delegate
+    status = mDelegate.ManuallyTriggerTransport(connectionID, activationReason, timeControl);
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void PushAvStreamTransportServer::HandleFindTransport(HandlerContext & ctx,
+                                                      const Commands::FindTransport::DecodableType & commandData)
+{
+    Status status = Status::Success;
+    Commands::FindTransportResponse::Type response;
+
+    Optional<DataModel::Nullable<uint16_t>> connectionID = commandData.connectionID;
+
+    DataModel::List<const TransportConfigurationStruct> outTransportConfigurations;
+
+    if (connectionID.Value().IsNull())
+    {
+        if (mCurrentConnections.size() == 0)
+        {
+            ChipLogError(Zcl, "HandleFindTransport: ConnectionID not found");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+            return;
+        }
+    }
+    else if (!FindStreamTransportConnection(connectionID.Value().Value()))
+    {
+        ChipLogError(Zcl, "HandleFindTransport: ConnectionID not found");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    // Call the delegate
+    status = mDelegate.FindTransport(connectionID, outTransportConfigurations);
```

Comment:
We should do fabric scoped searching here, but we don't pass fabric Index for this function.